packages/headless/src/controllers/insight/facets/range-facet/date-facet/headless-insight-date-facet.ts

```diff
@@ -0,0 +1,77 @@
+import {DateRangeRequest} from '../../../../../features/facets/range-facets/date-facet-set/interfaces/request';
+import {DateFacetValue} from '../../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+
+import {getAnalyticsActionForToggleRangeFacetSelect} from '../../../../../features/facets/range-facets/generic/range-facet-utils';
+import {DateFacetOptions} from '../../../../core/facets/range-facet/date-facet/headless-date-facet-options';
+import {
+  buildCoreDateFacet,
+  buildDateRange,
+  DateFacet,
+  DateFacetProps,
+  DateFacetState,
+  DateRangeInput,
+  DateRangeOptions,
+} from '../../../../core/facets/range-facet/date-facet/headless-core-date-facet';
+import {
+  logFacetClearAll,
+  logFacetUpdateSort,
+} from '../../../../../features/facets/facet-set/facet-set-analytics-actions';
+import {RangeFacetSortCriterion} from '../../../../../features/facets/range-facets/generic/interfaces/request';
+import {InsightEngine} from '../../../../../app/insight-engine/insight-engine';
+import {executeSearch} from '../../../../../features/insight-search/insight-search-actions';
+
+export type {
+  DateFacetOptions,
+  DateRangeInput,
+  DateRangeOptions,
+  DateRangeRequest,
+  DateFacetProps,
+  DateFacet,
+  DateFacetState,
+};
+export {buildDateRange};
+
+/**
+ * Creates a `DateFacet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `DateFacet` controller properties.
+ * @returns A `DateFacet` controller instance.
+ */
+export function buildInsightDateFacet(
+  engine: InsightEngine,
+  props: DateFacetProps
+): DateFacet {
+  const coreController = buildCoreDateFacet(engine, props);
+  const dispatch = engine.dispatch;
+  const getFacetId = () => coreController.state.facetId;
+
+  return {
+    ...coreController,
+
+    deselectAll() {
+      coreController.deselectAll();
+      dispatch(executeSearch(logFacetClearAll(getFacetId())));
+    },
+
+    sortBy(criterion: RangeFacetSortCriterion) {
+      coreController.sortBy(criterion);
+      dispatch(
+        executeSearch(logFacetUpdateSort({facetId: getFacetId(), criterion}))
```

Comment:
we need to change this event with proper client, in our case `CoveoInsightClient`

Reply:
This PR add the new analytics actions https://github.com/coveo/ui-kit/pull/2147